jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/table/IGridAssert.java

```diff
@@ -0,0 +1,231 @@
+     * @param columns to compare
+     */
+    @JDIAction("Assert that '{name}' has columns '{0}'")
+    public A columns(List<String> columns) {
+        for(String column : columns)
+            column(column);
+        return (A) this;
+    }
+
+    /**
+     * Match passed value with table columns
+     * @param condition to compare
+     */
+    @JDIAction("Assert that '{name}' columns '{0}'")
+    public A columns(Matcher<Collection<? extends String>> condition) {
+        jdiAssert(grid().header(), condition);
+        return (A) this;
+    }
+
+    @JDIAction("Assert that '{name}' row '{0}' equals to other row")
+    public A rowVisualValidation(String rowName, Line row) {
+        jdiAssert(grid().row(rowName).visualCompareTo(row), Matchers.is(true));
+        return (A) this;
+    }
+    private Line findRow(List<Line> rows, String name, String columnName) {
+        Line line = first(rows, l -> l.get(columnName).equals(name));
+        if (line == null)
+            throw exception("Can't find %s row with column %s", name, columnName);
+        return line;
+    }
+    @JDIAction("Assert that '{name}' is sorted by '{0}' in ascending order")
+    public A sortedByAsc(String columnName) {
+        List<String> column = grid().webColumn(columnName).values();
+        for (int i = 1; i < column.size(); i++)
+            if (!isSorted(column.get(i-1), column.get(i), true, false))
+                jdiAssert("Table is not by ascending at "+i+" row", Matchers.is(""));
+        return (A)this;
+    }
+    @JDIAction("Assert that '{name}' is sorted by '{0}' in descending order")
+    public A sortedByDesc(String columnName) {
+        List<String> column = grid().webColumn(columnName).values();
+        for (int i = 1; i < column.size(); i++)
+            if (!isSorted(column.get(i), column.get(i-1), true, false))
+                jdiAssert("Table is not by descending at "+i+" row", Matchers.is(""));
+        return (A)this;
+    }
+    @JDIAction("Assert that '{name}' is sorted by '{0}' in ascending order")
+    public A sortedNumByAsc(String columnName) {
+        List<String> column = map(grid().webColumn(columnName).values(), el -> el.replaceAll("[^0-9.,]", ""));
+        for (int i = 1; i < column.size(); i++)
+            if (!isSorted(column.get(i-1), column.get(i), true, false))
+                jdiAssert("Table is not by ascending at "+i+" row", Matchers.is(""));
+        return (A)this;
+    }
+    @JDIAction("Assert that '{name}' is sorted by '{0}' in descending order")
+    public A sortedNumByDesc(String columnName) {
+        List<String> column = map(grid().webColumn(columnName).values(), el -> el.replaceAll("[^0-9.,+\\-]", ""));
+        for (int i = 1; i < column.size(); i++)
+            if (!isSorted(column.get(i), column.get(i-1), true, false))
+                jdiAssert("Table is not by descending at "+i+" row", Matchers.is(""));
+        return (A)this;
+    }
+
+     private Class<?> cl;
```

Comment:
Codacy found an issue: [Perhaps 'cl' could be replaced by a local variable.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6397461)